app/webhook.go

```diff
@@ -95,21 +95,22 @@ func (a *App) handleWebhookEvents(c request.CTX, post *model.Post, team *model.T
 }
 
 func (a *App) TriggerWebhook(c request.CTX, payload *model.OutgoingWebhookPayload, hook *model.OutgoingWebhook, post *model.Post, channel *model.Channel) {
-	var body io.Reader
-	var contentType string
-	if hook.ContentType == "application/json" {
-		js, err := json.Marshal(payload)
-		if err != nil {
-			c.Logger().Warn("Failed to encode to JSON", mlog.Err(err))
-		}
-		body = bytes.NewReader(js)
-		contentType = "application/json"
-	} else {
-		body = strings.NewReader(payload.ToFormValues())
-		contentType = "application/x-www-form-urlencoded"
-	}
 
 	for i := range hook.CallbackURLs {
+		var body io.Reader
+		var contentType string
+		if hook.ContentType == "application/json" {
```

Comment:
From looking at this code change, it seems the issue is that we try to reuse the same reader (in this case the `body` variable). Creating a new reader per callback URL makes sense.

I'm thinking we may want to only marshal the payload once, mainly for readability. @hanzei Do you think this is worth this change?

```suggestion
	var jsonBytes []byte
	var err error

	contentType := "application/x-www-form-urlencoded"
	if hook.ContentType == "application/json" {
		contentType = "application/json"
		jsonBytes, err = json.Marshal(payload)
		if err != nil {
			c.Logger().Warn("Failed to encode to JSON", mlog.Err(err))
			return
		}
	}

	for i := range hook.CallbackURLs {
		var body io.Reader
		if hook.ContentType == "application/json" {
			body = bytes.NewReader(jsonBytes)
		} else {
			body = strings.NewReader(payload.ToFormValues())
		}
```

Reply:
Good idea. Fixed it